### PR TITLE
docs(troubleshooting): add troubleshooting documentation

### PR DESCRIPTION
**Pull request by JamesSingleton**
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: docs/troubleshooting/error-while-attempting-to-call-load-module-data.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Error While Attempting to Call 'load' or 'loadModuleData' Inside Holocron Module
```
Comment (JamesSingleton) [inline docs/troubleshooting/error-while-attempting-to-call-load-module-data.md:1]: I wasn't sure what would be the best `Possible Ways to Fix It`

**Review by Francois-Esquire — DISMISSED** [review]

File: docs/troubleshooting/error-while-attempting-to-call-load-module-data.md (hunk 1 above)
Comment (infoxicator) [reply docs/troubleshooting/error-while-attempting-to-call-load-module-data.md:1]: this is usually an uncaught error in the `loadModuleData` promise chain that bubbles up and gets caught by One App. The way to debug it depends on the function you are executing there. suggesting adding try catches to debug the actual error maybe?

File: docs/troubleshooting/module-does-not-exist-in-module-map.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Could Not Load Module Because It Does Not Exist in the Module Version Map
+
+#### Why This Error Occurred
+
+Sometimes you might want to compose a module from multiple smaller modules using either [iguazu-holocron](https://github.com/americanexpress/holocron/tree/main/packages/iguazu-holocron) or [RenderModule](https://github.com/americanexpress/holocron/blob/main/packages/holocron/docs/api/README.md#rendermodule). However, if you have not added the module you are trying to load to your [module-map](../api/server/Module-Map-Schema), it will fail to load.
```
Comment (infoxicator) [inline docs/troubleshooting/module-does-not-exist-in-module-map.md:5]: I think this error also occurs if you use `RenderModule` without dispatching `composeModules` in `LoadModuleData` could you please check if that's the case add it here?

File: docs/troubleshooting/refused-to-connect-to-url-due-to-csp-violation.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+#### Why This Error Occurred
+
+One App requires that you set a [Content Security Policy (CSP)](../api/modules/App-Configuration#csp) within your root module. Not having this set properly will result in One App refusing to connect to a URL.
+
+#### Possible Ways to Fix It
+
+The error in your browser console will include where the URL you are trying to connect to needs to be added. For example:
+
+```sh
+violates the following Content Security Policy directive: "connect-src 'self' *.example.com".
```
Comment (infoxicator) [inline docs/troubleshooting/refused-to-connect-to-url-due-to-csp-violation.md:12]: We probably need a recipe on how to configure the CSP

**Review by infoxicator — CHANGES_REQUESTED** [review]

File: docs/troubleshooting/refused-to-connect-to-url-due-to-csp-violation.md (hunk 3 above)
Comment (JamesSingleton) [reply docs/troubleshooting/refused-to-connect-to-url-due-to-csp-violation.md:12]: Probably, we can tackle that in another PR

Comment (infoxicator) [reply docs/troubleshooting/refused-to-connect-to-url-due-to-csp-violation.md:12]: Oh yeah, didn't mean adding it here.

File: docs/troubleshooting/module-does-not-exist-in-module-map.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Furthermore, if you are calling `RenderModule` but not calling `composeModules` in `loadModuleData`, this will result in a warning stating that the module is not in the module map.
```
Comment (infoxicator) [inline docs/troubleshooting/module-does-not-exist-in-module-map.md:7]: can you link to the holocron docs that explain how to use `composeModules`

**Review by infoxicator — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**